Test_Automation/library/danos_cli.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/python3
+# * Copyright (c) 2020-2021, Happiest Minds Technologies Limited Intellectual Property. All rights reserved.
+# *
+# * SPDX-License-Identifier: LGPL-2.1-only
+import time
+import vymgmt
+
+def config_set(ip, user, ps, cmd):
+    handle = vymgmt.Router(ip, user, password=ps, port=22)
+    handle.login()
+    handle.configure()
+    handle.set(cmd)
+    handle.commit()
+    handle.save()
+    handle.exit()
+    handle.logout()
+
+def config_show_service(ip, user, ps, cmd):
+    handle = vymgmt.Router(ip, user, password=ps, port=22)
+    handle.login()
+    handle.configure()
+    out = handle.run_conf_mode_command(cmd)
+    handle.exit()
+    handle.logout()
+    output = out.split("\n")
+    output2 = ''.join([str(elem) for elem in output])
+    return output2
+
+def config_delete(ip, user, ps, cmd):
+    handle = vymgmt.Router(ip, user, password=ps, port=22)
+    handle.login()
+    handle.configure()
+    handle.delete(cmd)
+    handle.commit()
+    handle.save()
+    handle.exit()
+    handle.logout()
+
+def show_command(ip, user, ps, cmd):
+    handle = vymgmt.Router(ip, user, password=ps, port=22)
+    handle.login()
+    out = handle.run_op_mode_command(cmd)
+    handle.exit()
+    handle.logout()
+    output = out.split("\n")
+    #output2 = ''.join([str(elem) for elem in output])
+    return output
+
+def config_ipsecvpn(ip, user, ps, cmd):
+    handle = vymgmt.Router(ip, user, password=ps, port=22)
+    handle.login()
+    handle.configure()
+    for line in cmd:
+        print(line)
+        handle.set(line)
+    handle.commit()
+    handle.save()
+    handle.exit()
+    handle.logout()
+
+def config_mplsldp(ip, user, ps, cmd):
```

Comment:
How does this differ from 'config_ipsecvpn' or indeed 'config'?

Would it be better to have a general configuration (your config function) function that can configure anything? This way you can improve your configuration handler in one place - and I think it would be less confusing to other developers.

Reply:
Yes. Agree on this and this will be the best practice.
The only reason for creating this function is that multiple commands used for configuring (basically, list is sent) and internally used a loop to achieve the configuration of all commands.
The other method was written already for a single command execution was used with that for differ scripts.
will have a look into this later.

Reply:
Hi Trumpton,
I have used completely general methods in the new script which are added for BGP test scenarios. Please have a look the same and provide your comments

Reply:
Will take care of the other scripts too so that the dependency of python library is avoided now.